ConwaysGameOfLife/GameController.cs

```diff
@@ -76,51 +84,61 @@ public void Run()
                         _output.WriteLine($"or {Messages.Play}");
 
                         input = _input.ReadLine();
-
-                        // if (UserEndsGame(input))
+            
+                        if (UserEndsGame(input))
+                        {
+                            _output.WriteLine(Messages.GameEnd);
+                            return;
+                        }
+                        // if (UserPressesPToPlay(input))
                         // {
-                        //     _output.WriteLine(Messages.GameEnd);
                         //     return;
                         // }
-
                         var isValidLocation = Validator.IsValidLocation(input, _universe.GridWidth, _universe.GridLength);
                         while (!isValidLocation)
                         {
                             _output.WriteLine(Messages.InvalidInput);
                             _output.WriteLine($"{Messages.RequestLiveCell}.");
                             input = _input.ReadLine();
 
-                            // if (UserEndsGame(input))
+                            if (UserEndsGame(input))
+                            {
+                                _output.WriteLine(Messages.GameEnd);
+                                return;
+                            }
+
+                            // if (UserPressesPToPlay(input))
                             // {
-                            //     _output.WriteLine(Messages.GameEnd);
                             //     return;
                             // }
+                            
                             isValidLocation = Validator.IsValidLocation(input, _universe.GridWidth, _universe.GridLength);
                         }
                         var location = InputParser.ParseLocation(input);
 
                         SetLiveCellLocation(location);
                         DisplayUniverse();
                     }
-                    while (input != "p");
+                    while (!UserPressesPToPlay(input));
                     Play();
-                }
-            }
-                while (_input.ReadKey(true).Key != ConsoleKey.Q);
+            //     }
+            // }
+            // while (_input.ReadKey(true).Key != ConsoleKey.Q);
         }
 
         private void Play()
         {
-            //var generator = new Generator();
-
-            // loop until all cells are dead or user enters q
-            var generator = new Generator(_universe);
-            generator.GenerateNewUniverse();
+            while(!_universe.AreAllCellsDead())
+            {
+                var generator = new Generator(_universe);
+                generator.GenerateNewUniverse();

```

Comment:
Not saving the generated `Universe` in any way here
```suggestion
                _universe = generator.GenerateNewUniverse();
```